torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1354,6 +1357,31 @@ void ProcessGroupNCCL::workCleanupLoop() {
     // Bump up heart beat by one.
     heartbeat_++;
 
+    // poll store to see if some ranks have flagged a timeout when
+    // we haven't polled for `heartbeat_timeout` seconds and there haven't
+    // any work added or removed for `watchdog_timeout` seconds.
+    if (dumpOnTimeout_) {
+      auto currentTime = std::chrono::steady_clock::now();
+      auto timeSinceLastWorkListUpdate =
+          (currentTime - lastWorkListUpdateTime_).count();
+      auto timeSinceLastPollStore = (currentTime - lastTimePollStore).count();
+      if (timeSinceLastWorkListUpdate >= kWatchdogThreadSleepMillis &&
+          timeSinceLastPollStore >= heartbeatTimeoutInSec_ * 1000) {
+        lastTimePollStore = currentTime;
+        if (store_->check({std::string(TIMEOUT_DUMP)}) && !optAsyncDebugDump) {
+          optAsyncDebugDump = launchAsyncDebugDump();
+          optAsyncDebugDump->wait_for(
+              std::chrono::milliseconds(kWatchdogThreadSleepMillis * 30));
+          const auto exitMsg = c10::str(
+              "[Rank ",
+              rank_,
+              "] NCCL watchdog thread detected timeout from Store");
```

Comment:
this message could be clearer.  (think of a user, they don't know what Store is or why it would be timing out.

More helpful would be to explain that some other rank's watchdog detected a timeout and notified us so we're dumping and aborting.

Reply:
sure.. That is a good point, let me change it.